### PR TITLE
Update E2E tests to use PHPUnit 8

### PR DESCRIPTION
**Pull request by sanmai**
This PR:

- [x] Updates E2E tests to use PHPUnit 8

### CODE REVIEW

**Comment by sanmai** [comment]
There's something off with `PhpUnit_Depends_Order`, will be looking into it.

File: tests/Fixtures/e2e/PhpUnit_Depends_Order/expected-output.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Total: 1
-Killed: 0
-Errored: 1
+Killed: 1
+Errored: 0
```
Comment (sanmai) [inline tests/Fixtures/e2e/PhpUnit_Depends_Order/expected-output.txt]: @maks-rafalko can you tell why there was an erroring mutation?

Comment (sanmai) [reply tests/Fixtures/e2e/PhpUnit_Depends_Order/expected-output.txt]: I get this if I downgrade back again:
```
PHPUnit 6.5.14 by Sebastian Bergmann and contributors.

E
Fatal error: Uncaught Exception: One of the tests were skipped. Something wrong with the order of executed tests.
``` 
Is this all right that we stopped getting this error?

Comment (sanmai) [reply tests/Fixtures/e2e/PhpUnit_Depends_Order/expected-output.txt]: As I see it this E2E test only supposed to test the initial run with a random order, nothing else.

**Comment by sanmai** [comment]
AppVeyor is failing with "Unable to locate a Composer executable on local system. Ensure that Composer is installed and available."

**Comment by maks-rafalko** [comment]
> @maks-rafalko can you tell why there was an erroring mutation?

Because it was an error in the log. 

Before:

```bash
Escaped mutants:
================

Timed Out mutants:
==================

Killed mutants:
===============

Errors mutants:
===============


1) /infection/tests/Fixtures/e2e/PhpUnit_Depends_Order/src/SourceClass.php:7    [M] PublicVisibility
'/infection/tests/Fixtures/e2e/PhpUnit_Depends_Order/vendor/phpunit/phpunit/phpunit' '--configuration' '/infection/tests/Fixtures/e2e/PhpUnit_Depends_Order/./infection/phpunitConfiguration.f9968f13232bd9033816281599b1829e.infection.xml'
--- Original
+++ New
@@ @@
 
 class SourceClass
 {
-    public function hello() : string
+    protected function hello() : string
     {
         return 'hello';
     }
 }
PHPUnit 7.3.0 by Sebastian Bergmann and contributors.

E
Fatal error: Uncaught Exception: One of the tests were skipped. Something wrong with the order of executed tests. in /infection/tests/Fixtures/e2e/PhpUnit_Depends_Order/tests/SourceClassTest.php:27
Stack trace:
#0 /infection/tests/Fixtures/e2e/PhpUnit_Depends_Order/vendor/phpunit/phpunit/src/Framework/TestSuite.php(755): PhpUnit_Depends_Order\Test\SourceClassTest::tearDownAfterClass()
#1 /infection/tests/Fixtures/e2e/PhpUnit_Depends_Order/vendor/phpunit/phpunit/src/Framework/TestSuite.php(750): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#2 /infection/tests/Fixtures/e2e/PhpUnit_Depends_Order/vendor/phpunit/phpunit/src/TextUI/TestRunner.php(587): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#3 /infection/tests/Fixtures/e2e/PhpUnit_Depends_Order/vendor/phpunit/phpunit/src/TextUI/Command.php(203): PHPUnit\TextUI\TestRunner->doRun(Object(PHPUnit\Framework\ in /infection/tests/Fixtures/e2e/PhpUnit_Depends_Order/tests/SourceClassTest.php on line 27

Not Covered mutants:
====================
```

And it was ok (IMO).

But now something has changed in PHPUnit as we get a different ouput for the same e2e script

```bash
Escaped mutants:
================

Timed Out mutants:
==================

Killed mutants:
===============


1) /infection/tests/Fixtures/e2e/PhpUnit_Depends_Order/src/SourceClass.php:7    [M] PublicVisibility
'/infection/tests/Fixtures/e2e/PhpUnit_Depends_Order/vendor/phpunit/phpunit/phpunit' '--configuration' '/infection/tests/Fixtures/e2e/PhpUnit_Depends_Order/./infection/phpunitConfiguration.68010bbcfc18dfe222fe48dfd3e05470.infection.xml'
--- Original
+++ New
@@ @@
 
 class SourceClass
 {
-    public function hello() : string
+    protected function hello() : string
     {
         return 'hello';
     }
 }
PHPUnit 8.2.4 by Sebastian Bergmann and contributors.

EF

Time: 89 ms, Memory: 4.00 MB

There was 1 error:

1) PhpUnit_Depends_Order\Test\SourceClassTest::test_hello
Error: Call to protected method PhpUnit_Depends_Order\SourceClass::hello() from context 'PhpUnit_Depends_Order\Test\SourceClassTest'

/infection/tests/Fixtures/e2e/PhpUnit_Depends_Order/tests/SourceClassTest.php:34

--

There was 1 failure:

1) PhpUnit_Depends_Order\Test\SourceClassTest::tearDownAfterClass
Exception in PhpUnit_Depends_Order\Test\SourceClassTest::tearDownAfterClass
One of the tests were skipped. Something wrong with the order of executed tests.

/infection/tests/Fixtures/e2e/PhpUnit_Depends_Order/tests/SourceClassTest.php:27

ERRORS!
Tests: 2, Assertions: 0, Errors: 1, Failures: 1.

Errors mutants:
===============

Not Covered mutants:
====================
```

This output is no longer treated as "error" in infection. 

The exit code is `2` on these lines:

https://github.com/infection/infection/blob/f4a72a83152c32551d3311f180d21d2183b49ee1/src/Process/MutantProcess.php#L117-L119 and condition is falsy.

But for PHPUnit < 8 it was `255`.

**Pull request merged**